robots/EDUCATOR/gyro.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+"""
+Move robot in a square path using the Gyro sensor.
+
+This script is a simple demonstration of turning using the Gyro sensor.
+"""
+
+from ev3dev.motor import MoveSteering, OUTPUT_B, OUTPUT_C
+from ev3dev.sensor.lego import GyroSensor
+from time import sleep
+
+
+class GyroSensorEnhanced(GyroSensor):
+    """Enhanced Gyro."""
+
+    def wait_until_changed_by(self, delta):
+        """Wait until angle has changed by specified amount."""
+        start_angle = self.value()
+        while abs(start_angle - self.value()) < delta:
+            sleep(0.01)
+
+motor_pair = MoveSteering(OUTPUT_B, OUTPUT_C)
+gyro = GyroSensorEnhanced()
+gyro.mode = GyroSensor.MODE_GYRO_ANG
+
+for i in range(4):
+    # Move robot forward for 3 seconds
+    motor_pair.on_for_seconds(steering=0, speed_pct=50, seconds=3)
+    # Spin robot to the left
+    motor_pair.on(steering=-100, speed_pct=5)
+    # Wait until angle changed by 90 degrees
+    gyro.wait_until_changed_by(90)
```

Comment:
This is cool :) How accurate of a square does it make?

Reply:
It’s reasonably accurate so long as the robot is not spinning too fast. It seems to work as well as the EV3-G equivalent. I’ve found the gyro sensor tends to lose accuracy as the batteries lose charge. Also, very occasionally the gyro sensor readings will continuously increment without any angular movement. This is nothing to do with ev3dev though. The same behaviour occurs with EV3-G.